File: TriangleChecker.py
#*******************************************************************
# TriangleChecker.py              Author: Caleb Shimomura
# Checks whether three integers supplied by the user can make
# a triangle in at least one configuration.
#
#*******************************************************************


#sides = input("Enter three numbers to see if they can make a triangle:")
#print(len(sides))
#print(sum(sides))

#find min number, find max number, if max-min>middle, no triangle

a = int(input("Enter the length of the first side: "))
while a < 1:
    a = int(input("Enter the length of the first side: "))
b = int(input("Enter the length of the second side: "))
while b < 1:
    b = int(input("Enter the length of the second side: "))
c = int(input("Enter the length of the third side: "))
while c < 1:
    c = int(input("Enter the length of the third side: "))

d=[a,b,c]
maxList = max(d)
d.remove(maxList)
minList = min(d)
d.remove(minList)

if maxList-minList < sum(d):
    print(a, " and ", b, " and ", c, " make a valid triangle.")
else:
    print(a, " and ", b, " and ", c, " do not make a valid triangle.")

File: TriangleChecker.cpp
//*******************************************************************
// TriangleChecker.cpp               Author: Caleb Shimomura
// A C++ program to check if 3 lengths can compose a triangle.
//
//*******************************************************************

#include <iostream>
using namespace std;

int main() {
   int num1;
   int num2;
   int num3;
   //get values
   
   cout << "Enter the three triangle side lengths. \n" ;
   cout << "First side length: ";
   cin >> num1;
   cout << "Second side length: ";
   cin >> num2;
   cout << "Third side length: ";
   cin >> num3;
   
   //if 1 is bigger than 2 and 3, add 2 and 3 together and compare to 1
   //if 2 is biggest, add 1 and 3 together and compare to 2
   //if 3 is biggest, add 1 and 2 together and compare to 1
   if (num1 > num2) {
      if (num1 > num3) {
         if (num2 + num3 > num1) {
            cout << "The side lengths " << num1 << " and " << num2 << " and " << num3 << " compose a valid triangle.";
         } else {
            cout << "The side lengths " << num1 << " and " << num2 << " and " << num3 << " do not compose a valid triangle.";
         }
      } else {
         if (num2 + num1 > num3) {
            cout << "The side lengths " << num1 << " and " << num2 << " and " << num3 << " compose a valid triangle.";
         } else {
            cout << "The side lengths " << num1 << " and " << num2 << " and " << num3 << " do not compose a valid triangle.";
         }
      }
   } else if (num2 > num3) {
      if (num1 + num3 > num2) {
            cout << "The side lengths " << num1 << " and " << num2 << " and " << num3 << " compose a valid triangle.";
      } else {
            cout << "The side lengths " << num1 << " and " << num2 << " and " << num3 << " do not compose a valid triangle.";      }
   } else {
      if (num2 + num1 > num3) {
            cout << "The side lengths " << num1 << " and " << num2 << " and " << num3 << " compose a valid triangle.";
      } else {
            cout << "The side lengths " << num1 << " and " << num2 << " and " << num3 << " do not compose a valid triangle.";
      }
   }

   return 0;
}
